### PR TITLE
Added mixins for class-based views that correctly attribute spans

### PR DESCRIPTION
**Pull request by brett-lempereur**
This pull request includes mixins that address the visibility issues with Django class-based views and Django Rest Framework viewsets identified in #324.  As an example, with Django Rest Framework and a `project.pizzastore.views` module:

```python
from ddtrace.contrib.django import mixins
from rest_framework.viewsets import ViewSet

class PizzaViewSet(mixins.ViewSetTraceMixin, ViewSet):
    pass
```

Would generate spans with the correct `project.pizzastore.views.PizzaViewSet.list`, `project.pizzastore.views.PizzaViewSet.create` resources, instead of aggregating all class-based view responses under the `project.pizzastore.views.PizzaViewSet` resource.  Equivalent spans `project.application.View.get`, `project.application.View.post` etc. would be generated with the `ddtrace.contrib.django.mixins.ViewTraceMixin` mixin.

These will not be automatically patched, but instead must be included as a mixin in view classes by developers.  It _may_ be possible to automate this using runtime method binding and inspecting the URL configuration to discover any class-based views, but this feels too unreliable and risky for my liking.

*Edit* Opening this for initial comments -- I'll add some unit tests tomorrow ;)

### CODE REVIEW

**Comment by palazzem** [comment]
Thank you very much @brett-lempereur ! I'm going to review it during the next week so that we can have it merged in the next minor release!

**Comment by palazzem** [comment]
@brett-lempereur because we're going to ship the 0.12, I'm moving this patch to another milestone (TBD). Let me know if you need help to bring this PR to the finishing line!

Thank you very much!

File: tests/commands/test_runner.py [hunk 1]
```diff
@@ -125,6 +125,16 @@ def test_priority_sampling_from_env(self):
             )
             assert out.startswith(b'Test success')
 
+    def test_priority_sampling_from_env(self):
```
Comment (jd) [inline tests/commands/test_runner.py]: This looks like a merge issue to me and should not be here.

**Comment by Kyle-Verhoog** [comment]
Our new django integration released in `v0.34` should provide better visibility into class-based views. I'm happy to reopen this if we would like to reconsider using this approach.

**Pull request closed without merging**